spec.html

```diff
@@ -5595,6 +5855,128 @@ <h1>CopyDataProperties ( _target_, _source_, _excludedItems_ )</h1>
     </emu-clause>
+
+    <emu-clause id="sec-newprivatename" aoid="NewPrivateName">
+      <h1>NewPrivateName ( _description_ )</h1>
+      <emu-alg>
+        1. Assert: Type(_description_) is String.
+        1. Return a new unique Private Name whose [[Description]] value is _description_.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-privatefieldfind" aoid="PrivateFieldFind">
+      <h1>PrivateFieldFind ( _P_, _O_ )</h1>
+      <emu-alg>
+        1. Assert: _P_ is a Private Name.
+        1. Assert: Type(_O_) is Object.
+        1. For each element _entry_ of _O_.[[PrivateFieldValues]], do
+          1. If _entry_.[[PrivateName]] is _P_, return _entry_.
+        1. Return ~empty~.
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-privatefieldadd" aoid="PrivateFieldAdd">
+      <h1>PrivateFieldAdd ( _P_, _O_, _value_ )</h1>
+      <emu-alg>
+        1. Assert: _P_ is a Private Name.
+        1. Assert: Type(_O_) is Object.
+        1. Let _entry_ be ! PrivateFieldFind(_P_, _O_).
+        1. If _entry_ is not ~empty~, throw a *TypeError* exception.
+        1. Append PrivateField { [[PrivateName]]: _P_, [[PrivateFieldValue]]: _value_ } to _O_.[[PrivateFieldValues]].
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-privatefieldget" aoid="PrivateFieldGet">
+      <h1>PrivateFieldGet ( _P_, _O_ )</h1>
+      <emu-alg>
+        1. Assert: _P_ is a Private Name.
+        1. Assert: Type(_O_) is Object.
+        1. If _P_.[[Kind]] is `"field"`, then
+          1. Let _entry_ be ! PrivateFieldFind(_P_, _O_).
+          1. If _entry_ is ~empty~, throw a *TypeError* exception.
+          1. Return _entry_.[[PrivateFieldValue]].
+        1. Perform ? PrivateBrandCheck(_O_, _P_).
+        1. If _P_.[[Kind]] is `"method"`, then
+          1. Return _P_.[[Value]].
+        1. Else,
+          1. Assert: _P_.[[Kind]] is `"accessor"`.
+          1. If _P_ does not have a [[Get]] field, throw a *TypeError* exception.
+          1. Let _getter_ be _P_.[[Get]].
+          1. Return ? Call(_getter_, _O_).
+      </emu-alg>
+    </emu-clause>
+
+    <emu-clause id="sec-privatefieldset" aoid="PrivateFieldSet">
+      <h1>PrivateFieldSet ( _P_, _O_, _value_ )</h1>
+      <emu-alg>
+        1. Assert: _P_ is a Private Name.
+        1. Assert: Type(_O_) is Object.
+        1. If _P_.[[Kind]] is `"field"`, then
+          1. Let _entry_ be ! PrivateFieldFind(_P_, _O_).
+          1. If _entry_ is ~empty~, throw a *TypeError* exception.
+          1. Set _entry_.[[PrivateFieldValue]] to _value_.
+          1. Return.
+        1. If _P_.[[Kind]] is `"method"`, throw a *TypeError* exception.
+        1. Else,
+          1. Assert: _P_.[[Kind]] is `"accessor"`.
+          1. Perform ? PrivateBrandCheck(_O_, _P_).
+          1. If _P_ does not have a [[Set]] field, throw a *TypeError* exception.
+          1. Let _setter_ be _P_.[[Set]].
+          1. Perform ? Call(_setter_, _O_, _value_).
+          1. Return.
```

Comment:
I would reorder and drop the "return"s:
```suggestion
        1. If _P_.[[Kind]] is `"method"`, throw a *TypeError* exception.
        1. If _P_.[[Kind]] is `"field"`, then
          1. Let _entry_ be ! PrivateFieldFind(_P_, _O_).
          1. If _entry_ is ~empty~, throw a *TypeError* exception.
          1. Set _entry_.[[PrivateFieldValue]] to _value_.
        1. Else,
          1. Assert: _P_.[[Kind]] is `"accessor"`.
          1. Perform ? PrivateBrandCheck(_O_, _P_).
          1. If _P_ does not have a [[Set]] field, throw a *TypeError* exception.
          1. Let _setter_ be _P_.[[Set]].
          1. Perform ? Call(_setter_, _O_, _value_).
```